app/docs/md/patterns/event-listeners.md

```diff
@@ -0,0 +1,191 @@
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+As a best practice, event listeners for web components should be added inside the `connectedCallback` function, since this method runs when the element is added to the DOM.
+
+Inside our `connectedCallback` method, we'll use the `querySelector` method on the custom element, to get the button inside our component. Then we will add a `click` listener to the button.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  connectedCallback () {
+    this.button = this.querySelector('button')
+    this.button.addEventListener('click', function (event) {
+      alert('I was clicked')
+    })
+  }
+
+  render ({ html, state }) {
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+That works great but if we truly do want to follow best practices we should clean up our event listeners when our component is removed from the DOM. We'll remove that event listener in the `disconnectedCallback` method. In order to do that we'll need to refactor our component slightly.
+
+1. Extract our click handler to a method named `clickHander`.
+1. Save a reference to the `button` element in the `constructor`.
+1. Bind `this` to the callback function.
+1. Add the event listener in `connectedCallback`.
+1. Remove the event listener in `disconnectedCallback`.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  handleClick (event) {
+    alert('I was clicked')
+  }
+
+  constructor () {
+    super()
+    this.button = this.querySelector('button')
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  connectedCallback () {
+    this.button.addEventListener('click', this.handleClick)
+  }
+
+  disconnectedCallback () {
+    this.button.removeEventListener('click', this.handleClick)
+  }
+
+  render ({ html, state }) {
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+<doc-callout level="info" mark="ℹ️">
+
+If the bind syntax offends you, then you can always use a public class field and an arrow function instead.
```

Comment:
```suggestion
If [the `bind` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind) offends you, then you can always use [a public class field and an arrow function](https://www.aleksandrhovhannisyan.com/blog/javascript-fields-vs-methods/#alternatives-for-this-binding) instead.
```

Not sure if these are the best links to use, but I reckon these terms could be new to a lot of folks, so having some kind of reference for these is likely helpful!